specifyweb/workbench/upload/treerecord.py

```diff
@@ -158,10 +158,54 @@ def disambiguate(self, disambiguation: DA) -> "ScopedTreeRecord":
 
     def get_treedefs(self) -> Set:
         return set([self.treedef])
+    
+    def rescope_tree_from_row(self, row: Row) -> Tuple["ScopedTreeRecord", Optional["WorkBenchParseFailure"]]:
+        tree_rank_model = get_treedefitem_model(self.name)
+        tree_node_model = getattr(models, self.name.lower().title())
+
+        # Create a mapping of rank names to their treedef IDs
+        ranks = {rank.rank_name: rank.treedef_id for rank in self.ranks.keys()}
+        rank_names = set(ranks.keys())
+
+        # Find non-null rank columns in the row
+        ranks_in_row_not_null = {
+            col
+            for col, value in row.items()
+            if value is not None and value != "" and col in rank_names
```

Comment:
Don't check for `value != ''`. it's already handled before you get here. It makes me think "do other places need this?"

Reply:
From debugging, `value != ""` was found to be needed.  An example  row value I found was:
```
{'FossilKingdom': '', 'Kingdom': 'test5', 'RockKingdom': '', 'Cat #': '#########', 'MineralKingdom': ''}
```